File: day6.py
grid = {}
gridSize = 1000

for i in xrange(0, gridSize):
    for n in xrange(0, gridSize):
        grid[i, n] = 0


def handle_instruction(instruction):

    _range_x = [0,0]
    _range_y = [0,0]

    _ranges = instruction\
        .replace("turn off ", "") \
        .replace("turn on ", "") \
        .replace("toggle ", "") \
        .replace("through ", "") \
        .split(" ")

    _range_x[0] = int(_ranges[0].split(',')[0])
    _range_x[1] = int(_ranges[1].split(',')[0])
    _range_y[0] = int(_ranges[0].split(',')[1])
    _range_y[1] = int(_ranges[1].split(',')[1])

    # At this point, we know the ranges.
    for _x in xrange(_range_x[0], _range_x[1] + 1):
        for _y in xrange(_range_y[0], _range_y[1] + 1):

            if instruction.startswith("turn off"):
                grid[_x, _y] = 0
            elif instruction.startswith("turn on"):
                grid[_x, _y] = 1
            else:
                # Toggle light
                if grid[_x, _y] == 1:
                    grid[_x, _y] = 0
                else:
                    grid[_x, _y] = 1


f = open('./input/day6.txt', 'r')
for line in f:

    # Trim input
    line = line.strip()

    # Handle the instruction set
    handle_instruction(line)

lightsOn = 0

for key in grid:
    if grid[key] == 1:
        lightsOn += 1

print str(lightsOn) + " lights turned on"
f.close()

File: day6.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            const int gridSize = 1000;

            // Initiate the grid
            int[,] grid = new int[gridSize + 1, gridSize + 1];

            for (int x = 0; x <= gridSize; x++)
                for (int y = 0; y <= gridSize; y++)
                    grid[x, y] = 0;

            // Keep track of the lights turned on
            int lightsOn = 0;

            // Read the file
            string line = string.Empty;

            StreamReader reader = new StreamReader("./input/day6.txt");
            while (!reader.EndOfStream)
                HandleInstruction(ref grid, reader.ReadLine());

            // check grid
            for (int x = 0; x <= gridSize; x++)
                for (int y = 0; y <= gridSize; y++)
                    if (grid[x, y] == 1)
                        lightsOn++;

            Console.WriteLine(lightsOn.ToString() + " lights turned on");
            Console.ReadKey();
        }

        static void HandleInstruction(ref int[,] grid, string instruction)
        {
            int[,] _range_x = new int[2, 2];
            int[,] _range_y = new int[2, 2];

            string[] _ranges = instruction
                .Replace("turn off ", "")
                .Replace("turn on ", "")
                .Replace("toggle ", "")
                .Replace("through ", "")
                .Split(' ');

            _range_x[0, 0] = int.Parse((_ranges[0].Split(',')[0]));
            _range_x[0, 1] = int.Parse((_ranges[1].Split(',')[0]));
            _range_y[0, 0] = int.Parse((_ranges[0].Split(',')[1]));
            _range_y[0, 1] = int.Parse((_ranges[1].Split(',')[1]));

            // At this point, we know the ranges.
            for (int x = _range_x[0,0]; x <= _range_x[0,1]; x++)
            {
                for (int y = _range_y[0, 0]; y <= _range_y[0, 1]; y++)
                {
                    if (instruction.StartsWith("turn off"))
                        grid[x, y] = 0;
                    else if (instruction.StartsWith("turn on"))
                        grid[x, y] = 1;
                    else
                        grid[x, y] = (grid[x, y] == 0) ? 1 : 0;
                }
            }
        }
    }
}

